### PR TITLE
Added dipoleMagnitude colvar. Docs updated also.

### PR DESCRIPTION
**Pull request by alejob**
Hi, I've implemented the dipole magnitude collective variable. It is similar to dipole angle collective variable, details in #41 

Please let me know if you need anything else.

### CODE REVIEW

**Comment by alejob** [comment]
I must solve a conflict in colvars-refman-main.tex

**Comment by giacomofiorin** [comment]
@alejob Thanks for including this new type of variable.  Unfortunately the version that you forked from precedes a reorganization of the reference manual (see https://github.com/Colvars/colvars/issues/79) and made the merge more difficult.

One question: in the doc, the variable `dipoleMagnitude` is listed by you among those with non-scalar values, but the function implemented returns a scalar, and the name suggests that you are using `$|\vec{d}|$` as a variable, not `$\vec{d}$` (3-dimensional vector).  Can you confirm this?

**Comment by alejob** [comment]
You're right, Giacomo, `dipoleMagnitude` should be in the scalar section. I'll update the docs to fix this mistake.

**Comment by alejob** [comment]
I think that now everything is ok.

File: src/colvarcomp_distances.cpp [hunk 1]
```diff
@@ -740,6 +740,63 @@ void colvar::distance_pairs::apply_force(colvarvalue const &force)
...
+    ai->grad = (ai->charge - aux1*ai->mass) * dipVunit;
+  }
+}
+
+void colvar::dipole_magnitude::apply_force (colvarvalue const &force)
+{
+  if (!atoms->noforce) {
+    atoms->apply_colvar_force (force.real_value);
+    for (cvm::atom_iter ai = atoms->begin(); ai != atoms->end(); ai++) {
+      cvm::atom_pos forceV=force.real_value*ai->grad;
```
Comment (giacomofiorin) [inline src/colvarcomp_distances.cpp]: @alejob Can you clarify the purpose of this line?

Comment (alejob) [reply src/colvarcomp_distances.cpp]: The lines 789 and 790 were to debugging, they built a force vector that was used to check if the total sum over the molecule was zero (don't move the center of mass). I'll check it again, and I'll do a new commit removing this lines. Thanks for the review.

Comment (giacomofiorin) [reply src/colvarcomp_distances.cpp]: No worries, I just wanted to make sure that the gradients are implemented as you intend them.

**Pull request merged**